docs/eventing/sources/containersource.md

```diff
@@ -0,0 +1,180 @@
+---
+title: "ContainerSource"
+linkTitle: "ContainerSource"
+weight: 31
+type: "docs"
+---
+
+![version](https://img.shields.io/badge/API_Version-v1beta1-red?style=flat-square)
+
+ContainerSource will start a container image which will generate events under
+certain situations and send messages to a sink URI. It also can be an easy way
+to support your own event sources in Knative. This guide shows how to configure
+ContainerSource as an event source for functions and summarizes guidelines for
+creating your own event source as a ContainerSource.
+
+### Prerequisites
+- Install [ko](https://github.com/google/ko)
+- Set `KO_DOCKER_REPO`
+ (e.g. `gcr.io/[gcloud-project]` or `docker.io/<username>`)
+- Authenticated with your `KO_DOCKER_REPO`
+- Install [`docker`](https://docs.docker.com/install/)
+
+## Installation
+
+The ContainerSource source type is enabled by default when you install Knative Eventing.
+
+## Example
+
+This example shows how the heartbeats container sends events to a Knative Service.
```

Comment:
Knative Service -> Kubernetes Service

(BTW the PingSource page is also wrong)

Reply:
or better: to a a Event Display Service